frontend/src/containers/AdminPanel/AdminPanel.js

```diff
@@ -67,60 +74,71 @@ class AdminPanel extends Component {
 
   render() {
     const { user, setJwt } = this.props;
-    const { activeTab, recipes, users } = this.state;
+    const {
+      activeTab, recipes, users, render,
+    } = this.state;
+    let renderContainer = false;
+    if (render) {
+      if (isNil(user)) renderContainer = <Redirect to="/" />;
+      if (!isNil(user)) {
+        renderContainer = !user.is_admin ? <Redirect to="/" />
```

Comment:
Maybe instead of this `renderContainer` thing, we can just directly `return` whatever we want to return? Something like:
```
if (render) {
  if (isNil(user)) {
    return (<Redirect to="/login" />);
  }

  if (!user.is_admin) {
    return (<Redirect to="/" />);
  }

  return (
    // long thing here
  )
}

return null;
```